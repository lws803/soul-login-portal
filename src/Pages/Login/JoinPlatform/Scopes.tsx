import React from 'react';
import { Box, Divider, Text, VStack } from '@chakra-ui/react';
import ScopeRow from './Scopes/ScoprRow';

export default function Scopes() {
  return (
    <Box
      borderRadius="md"
      borderStyle="solid"
      borderColor="soul.mutedGrey"
      borderWidth="1px"
      padding="1rem"
    >
      <VStack alignItems="left">
        <Text>
          Joining this platform will allow the platform and it&apos;s developers
          to:
        </Text>
        <Divider />
        <VStack alignItems="left" overflowY="auto" maxHeight="30vh">
          <ScopeRow title="Access your account information." />
          <ScopeRow title="Modify your account." />
          <ScopeRow title="Access your user connections." />
          <ScopeRow title="Modify your user connections." />
          <ScopeRow title="Access your membership on this platform." />
          <ScopeRow title="Modify your membership on this platform." />
        </VStack>
      </VStack>
    </Box>
  );
}
