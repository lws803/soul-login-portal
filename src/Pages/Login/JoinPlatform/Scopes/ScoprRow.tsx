import { CheckCircleIcon, NotAllowedIcon } from '@chakra-ui/icons';
import { HStack, Text } from '@chakra-ui/react';

export default function ScopeRow({ title, isNotChecked }: Props) {
  return (
    <HStack>
      {isNotChecked ? (
        <NotAllowedIcon aria-label="Scope not checked" />
      ) : (
        <CheckCircleIcon color="soul.green.200" aria-label="Scope checked" />
      )}
      <Text color="soul.mutedGrey">{title}</Text>
    </HStack>
  );
}

type Props = {
  title: string;
  isNotChecked?: boolean;
};
