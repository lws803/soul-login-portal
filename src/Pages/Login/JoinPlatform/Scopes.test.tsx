import React from 'react';
import { render } from '@testing-library/react';

import Scopes from './Scopes';

describe(Scopes, () => {
  it('renders', () => {
    const { container } = render(<Scopes />);
    expect(container).toMatchSnapshot();
  });
});
