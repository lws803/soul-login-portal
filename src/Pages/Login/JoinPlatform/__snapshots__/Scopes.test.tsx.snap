// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scopes renders 1`] = `
<div>
  <div
    class="css-do7e2g"
  >
    <div
      class="chakra-stack css-xtejj8"
    >
      <p
        class="chakra-text css-0"
      >
        Joining this platform will allow the platform and it's developers to:
      </p>
      <hr
        aria-orientation="horizontal"
        class="chakra-divider css-1upb9tn"
      />
      <div
        class="chakra-stack css-111m4e8"
      >
        <div
          class="chakra-stack css-84zodg"
        >
          <svg
            aria-label="Scope checked"
            class="chakra-icon css-1rcwe32"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12,0A12,12,0,1,0,24,12,12.014,12.014,0,0,0,12,0Zm6.927,8.2-6.845,9.289a1.011,1.011,0,0,1-1.43.188L5.764,13.769a1,1,0,1,1,1.25-1.562l4.076,3.261,6.227-8.451A1,1,0,1,1,18.927,8.2Z"
              fill="currentColor"
            />
          </svg>
          <p
            class="chakra-text css-1rfq100"
          >
            Access your account information.
          </p>
        </div>
        <div
          class="chakra-stack css-84zodg"
        >
          <svg
            aria-label="Scope not checked"
            class="chakra-icon css-onkibi"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8 0-1.85.63-3.55 1.69-4.9L16.9 18.31C15.55 19.37 13.85 20 12 20zm6.31-3.1L7.1 5.69C8.45 4.63 10.15 4 12 4c4.42 0 8 3.58 8 8 0 1.85-.63 3.55-1.69 4.9z"
              fill="currentColor"
            />
          </svg>
          <p
            class="chakra-text css-1rfq100"
          >
            Modify your account.
          </p>
        </div>
        <div
          class="chakra-stack css-84zodg"
        >
          <svg
            aria-label="Scope checked"
            class="chakra-icon css-1rcwe32"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12,0A12,12,0,1,0,24,12,12.014,12.014,0,0,0,12,0Zm6.927,8.2-6.845,9.289a1.011,1.011,0,0,1-1.43.188L5.764,13.769a1,1,0,1,1,1.25-1.562l4.076,3.261,6.227-8.451A1,1,0,1,1,18.927,8.2Z"
              fill="currentColor"
            />
          </svg>
          <p
            class="chakra-text css-1rfq100"
          >
            Access your user connections.
          </p>
        </div>
        <div
          class="chakra-stack css-84zodg"
        >
          <svg
            aria-label="Scope checked"
            class="chakra-icon css-1rcwe32"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12,0A12,12,0,1,0,24,12,12.014,12.014,0,0,0,12,0Zm6.927,8.2-6.845,9.289a1.011,1.011,0,0,1-1.43.188L5.764,13.769a1,1,0,1,1,1.25-1.562l4.076,3.261,6.227-8.451A1,1,0,1,1,18.927,8.2Z"
              fill="currentColor"
            />
          </svg>
          <p
            class="chakra-text css-1rfq100"
          >
            Modify your user connections.
          </p>
        </div>
        <div
          class="chakra-stack css-84zodg"
        >
          <svg
            aria-label="Scope checked"
            class="chakra-icon css-1rcwe32"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12,0A12,12,0,1,0,24,12,12.014,12.014,0,0,0,12,0Zm6.927,8.2-6.845,9.289a1.011,1.011,0,0,1-1.43.188L5.764,13.769a1,1,0,1,1,1.25-1.562l4.076,3.261,6.227-8.451A1,1,0,1,1,18.927,8.2Z"
              fill="currentColor"
            />
          </svg>
          <p
            class="chakra-text css-1rfq100"
          >
            Access your membership on this platform.
          </p>
        </div>
        <div
          class="chakra-stack css-84zodg"
        >
          <svg
            aria-label="Scope checked"
            class="chakra-icon css-1rcwe32"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12,0A12,12,0,1,0,24,12,12.014,12.014,0,0,0,12,0Zm6.927,8.2-6.845,9.289a1.011,1.011,0,0,1-1.43.188L5.764,13.769a1,1,0,1,1,1.25-1.562l4.076,3.261,6.227-8.451A1,1,0,1,1,18.927,8.2Z"
              fill="currentColor"
            />
          </svg>
          <p
            class="chakra-text css-1rfq100"
          >
            Modify your membership on this platform.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
